article/app/model/AutomaticFilters.scala

```diff
@@ -0,0 +1,26 @@
+package model
+
+import common.GuLogging
+import model.TopMentionEntity.TopMentionEntity
+
+object AutomaticFilters extends GuLogging {
+
+  def getAutomaticFilter(filter: Option[String]): Option[(TopMentionEntity, String)] = {
+    filter.flatMap { f =>
+      val filterEntity = f.split(":")
+      if (filterEntity.length == 2) {
+        val entityType = TopMentionEntity.withNameOpt(filterEntity(0).toUpperCase)
```

Comment:
If the entityType always needs to be upercased is it worth handling that in the withNameOpt function instead?